services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureRedirectUrisEnforcerExecutor.java

```diff
@@ -205,12 +193,34 @@ public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyExcep
                 if (client == null) {
                     throw invalidRedirectUri("Invalid parameter: clientId");
                 }
-                verifyRedirectUri(redirectUriParam, true);
+                if (isAuthFlowWithRedirectEnabled(client)) {
+                    verifyRedirectUri(redirectUriParam, true);
+                }
                 return;
             default:
         }
     }
 
+    private void verifyRedirectUris(ClientCRUDContext context) throws ClientPolicyException {
+        ClientRepresentation client = context.getProposedClientRepresentation();
+        if (isAuthFlowWithRedirectEnabled(client)) {
+            List<String> redirectUris = client.getRedirectUris();
+            if (redirectUris == null || redirectUris.isEmpty()) {
+                throw invalidRedirectUri(ERR_GENERAL);
+            }
+            verifyRedirectUris(client.getRootUrl(), redirectUris);
+            verifyPostLogoutRedirectUriUpdate(client);
+        }
+    }
+
+    private static boolean isAuthFlowWithRedirectEnabled(ClientModel client) {
+        return client.isStandardFlowEnabled() || client.isImplicitFlowEnabled();
+    }
+
+    private static boolean isAuthFlowWithRedirectEnabled(ClientRepresentation client) {
+        return (client.isStandardFlowEnabled() == null || client.isStandardFlowEnabled() == Boolean.TRUE) || client.isImplicitFlowEnabled() == Boolean.TRUE;
```

Comment:
Could you tell me why you do not write as follows?
```
return (client.isStandardFlowEnabled() == null || client.isStandardFlowEnabled() == Boolean.TRUE) || (client.isImplicitFlowEnabled() == null || client.isImplicitFlowEnabled() == Boolean.TRUE);
```

Reply:
That comes down to the question how KC behaves internally when these values are null. My assumption was that `isStandardFlowEnabled == null` means that the flow is enabled (default value is `true`) and that `isImplicitFlowEnabled == null` means that the flow is disables (default value is `false`). I don't know if that assumption is correct, though.